spec/requests/users/invitation_spec.rb

```diff
@@ -1,153 +1,186 @@
+            # Arrange
             friend1 = User.find_by(email: invitee1)
             friend2 = User.find_by(email: invitee2)
+
+            # Assert
             expect(friend1.invitation_token).not_to eq(friend2.invitation_token)
           end
         end
       end
 
       context 'when invalid params are given' do
-        let(:invalid_invite) { post :create, params: { user: { email: invalid_email } } }
+        let(:invalid_invite) {
+          post user_invitation_path, params: { user: { email: invalid_email } }
+        }
 
         it 're-renders the invitation form' do
+          # Act
           invalid_invite
+
+          # Assert
           expect(response).to redirect_to new_user_invitation_path
         end
 
         it 'does not create a new user for an invalid email' do
-          user_count = User.count
-          invalid_invite
-          expect(User.count).to eq(user_count)
+          expect { invalid_invite }.not_to change { User.count }
         end
       end
 
       context 'when both valid and invalid params are given' do
-        before(:each) do
-          post :create, params: { user: { email: "#{invitee1}, #{invalid_email}" } }
-        end
+        let(:valid_and_invalid_invite) {
+          post user_invitation_path, params: { user: { email: "#{invitee1}, #{invalid_email}" } }
+        }
 
         it 'only creates a new User for the valid email' do
-          newest_users = User.order(invitation_sent_at: :desc).last(2)
-          emails = []
-          newest_users.each { |u| emails << u.email }
-          expect(emails.sort).to include(invitee1)
-          expect(emails.sort).not_to include(invalid_email)
+          # Assume
+          emails = User.pluck(:email)
+          expect(emails).not_to include(invitee1)
+          expect(emails).not_to include(invalid_email)
+
+          # Act
+          valid_and_invalid_invite
+
+          # Assert
+          emails = User.pluck(:email)
+          expect(emails).to include(invitee1)
+          expect(emails).not_to include(invalid_email)
         end
 
         it 'only sends an invitation email to the valid email address' do
+          # Arrange
           emails = []
+
+          # Act
+          expect { valid_and_invalid_invite }.to change { Devise.mailer.deliveries.count }.by(1)
+
+          # Assert
           Devise.mailer.deliveries.each { |d| emails << d.to[0] }
-          expect(Devise.mailer.deliveries.count).to eq(1)
-          expect(emails.sort).to include(invitee1)
-          expect(emails.sort).not_to include(invalid_email)
+          expect(emails).to include(invitee1)
+          expect(emails).not_to include(invalid_email)
         end
       end
     end
   end
 
   describe '#update' do
-    include_context :logged_in_user
     let(:password) { 'passworD@99' }
-
-    before(:each) do
-      invite_one_friend
-    end
+    let(:name) { 'New Person' }
+    let(:invited_user) { User.invite!({ email: invitee1 }, user) }
```

Comment:
This fixed the failing test when I moved this over from controller_spec to request_spec since I needed the `raw_invitation_token` from the user itself. If I use a route directly instead of the model method, the `raw_invitation_token` would have been deleted at that point and I won't have it to reference in the params for the test.